docs/extend/update-b14.md

```diff
@@ -0,0 +1,98 @@
+# Updating For Beta 14
+
+This release brings a large chunk of breaking changes - hopefully the last chunk of this size.
+In order to prepare the codebase for the upcoming stable release, we decided it was time to modernize / upgrade / exchange some of the underlying JavaScript libraries that are used in the frontend.
+Due to the nature and size of these upgrades, we have to pass on some of the breaking changes to you, our extension developers.
+
+On the bright side, this overdue upgrade brings us closer to the conventions of best practices of [Mithril.js](https://mithril.js.org/), the mini-framework used for Flarum's UI.
+Mithril's 2.0 release sports a more consistent component interface, which should be a solid foundation for years to come.
+Where possible, we replicated old APIs, to ease the upgrade and give you time to do the full transition.
+Quite a few breaking changes remain, though - read more below.
+
+::: tip
+If you need help with the upgrade, our friendly community will gladly help you out either [on the forum](https://discuss.flarum.org/t/extensibility) or [in chat](https://flarum.org/chat/).
+:::
+
+To ease the process, we've clearly separated the changes to the frontend (JS) from those in the backend (PHP) below.
+If your extension does not change the UI, consider yourself lucky. :-)
+
+## Frontend (JavaScript)
+
+### Changes in Core
+
+*TODO: States, and other BC breaks*
+
+### Mithril 2.0: Concepts
+
+*TODO: Explain*
+
+- props -> attrs
+- vnodes
+- Component instances should not be stored
+
+### How to upgrade a component
+
+#### Required changes
+
+*TODO*
+
+- `view()` -> `view(vnode)`
+- Lifecycle
+  - `init()` -> `oninit(vnode)`
+  - `config()` -> Lifecycle hooks `oncreate(vnode)` / `onupdate(vnode)`
+  - `context.onunload()` -> `onremove()`
+  - `SubtreeRetainer` -> `onbeforeupdate()`
+- if present, `attrs()` method needs to be renamed -> convention `elementAttrs()`
+- building component with `MyComponent.component()` -> `children` is now second parameter instead of a named prop/attr (first argument) -> JSX preferred
+- Routing
+  - `m.route()` -> `m.route.get()`
+  - `m.route(name)` -> `m.route.set(name)`
+  - register routes with page class, not instance
+    - special case when passing props
+  - `<a href={url} config={m.route}>` -> `<a route={url}>`
+- AJAX
+  - `m.request({...})` -> `data:` key split up into `body:` and `params:`
+  - `m.deferred` -> native `Promise`
+- Redrawing
+  - `m.redraw(true)` -> `m.redraw.sync()`
+  - `m.redraw.strategy('none')` -> `e.redraw = false` in event handler
+  - `m.lazyRedraw()` -> `m.redraw()`
```

Comment:
These will need to be explained thoroughly. But for now I wanted to collect the steps here. Is there anything missing (here or in the `Optional` section below) related to the Mithril 2.0 upgrade?

Reply:
I'd add that you can no longer use a `children` attribute unless the component extracts it before passing it as like `{...attrs}` or something.